src/EnergyPlus/UnitarySystem.cc

```diff
@@ -8529,6 +8580,32 @@ namespace UnitarySystems {
                                                       HeatCoilLoad,
                                                       SupHeaterLoad,
                                                       CompressorONFlag);
+                        PartLoadRatio = HeatPLR;
+                    } else if (DataGlobals::DoCoilDirectSolutions && CoolingLoad && this->m_CoolingCoilType_Num == DataHVACGlobals::CoilDX_Cooling &&
+                               this->m_NumOfSpeedCooling > 1) {
+                        HeatPLR = 0.0;
+                        if (this->m_CoolingSpeedNum == 1) {
+                            this->m_CoolingCycRatio = (ZoneLoad - SensOutputOff) / (this->FullOutput[this->m_CoolingSpeedNum] - SensOutputOff);
+                            CoolPLR = this->m_CoolingCycRatio;
+                            this->m_CoolingSpeedRatio = 0.0;
+                        } else {
+                            this->m_CoolingCycRatio = 1.0;
+                            this->m_CoolingSpeedRatio = (ZoneLoad - this->FullOutput[this->m_CoolingSpeedNum - 1]) /
+                                                        (this->FullOutput[this->m_CoolingSpeedNum] - this->FullOutput[this->m_CoolingSpeedNum - 1]);
+                            CoolPLR = this->m_CoolingSpeedRatio;
+                        }
+                        this->calcUnitarySystemToLoad(state, AirLoopNum,
+                                                      FirstHVACIteration,
+                                                      CoolPLR,
+                                                      HeatPLR,
+                                                      OnOffAirFlowRatio,
+                                                      SensOutput,
+                                                      LatOutput,
+                                                      HXUnitOn,
+                                                      HeatCoilLoad,
+                                                      SupHeaterLoad,
+                                                      CompressorONFlag);
+                        PartLoadRatio = CoolPLR;
```

Comment:
How does this extra section fit into the grand scheme?

Reply:
@Myoldmopar The extra code is used for a multiple speed coil to recalculate output based on cycleratio (SpeedNumber = 1) or SpeedRatio (SpeedNumber>1) using Direct solution results.